galaxyproject/galaxy
# Fix job directory not being cleaned up

Fixes:

```
galaxy.jobs ERROR 2024-10-15 13:11:57,570 [pN:main.1,p:610368,tN:LocalRunner.work_thread-2] Unable to cleanup job 11703
Traceback (most recent call last):
  File "/home/dlopez/dev/galaxy/lib/galaxy/jobs/__init__.py", line 2160, in cleanup
    self.object_store.delete(self.get_job(), base_dir="job_work", entire_dir=True, obj_dir=True)
  File "/home/dlopez/dev/galaxy/lib/galaxy/objectstore/__init__.py", line 580, in delete
    return self._invoke(
           ^^^^^^^^^^^^^
  File "/home/dlopez/dev/galaxy/lib/galaxy/objectstore/__init__.py", line 485, in _invoke
    return self.__getattribute__(f"_{delegate}")(obj=obj, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/dlopez/dev/galaxy/lib/galaxy/objectstore/__init__.py", line 1202, in _delete
    return self._call_method("_delete", obj, False, False, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/dlopez/dev/galaxy/lib/galaxy/objectstore/__init__.py", line 1524, in _call_method
    return self._resolve_backend(object_store_id).__getattribute__(method)(obj, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/dlopez/dev/galaxy/lib/galaxy/objectstore/__init__.py", line 1068, in _delete
    path = self._get_filename(obj, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/dlopez/dev/galaxy/lib/galaxy/objectstore/__init__.py", line 1114, in _get_filename
    raise ObjectNotFound
galaxy.exceptions.ObjectNotFound: No such object found.
```

## How to test the changes?
  - Run any tool
  - Check the server log and observe there is no exception when cleaning the job directory and it is indeed deleted.

## License
